src/main/scala/io/joern/typestubs/ruby/BuiltinPackageDownloader.scala

```diff
@@ -117,68 +119,46 @@ class BuiltinPackageDownloader(rubyVersion: String = "3.3.0") {
       .iterator
   }
 
+  /** Write RubyType for builtin types to MessagePack files and zipping directory
+    * @param rubyTypesMap
+    *   \- Map(Gemname -> List[RubyTypes for Gem])
+    */
   private def writeToFile(rubyTypesMap: collection.mutable.Map[String, List[RubyType]]): Unit = {
     val dir = File(s"${baseDir}/")
     dir.createDirectoryIfNotExists()
 
     rubyTypesMap.foreach { (gem, rubyTypes) =>
-      // gem is file name
-      val gemsMap = collection.mutable.Map[String, List[RubyType]]()
-
-      rubyTypes.foreach { rubyType =>
-        val rubyTypeNameSegments = rubyType.name.split("\\.")
-
-        val namespaceKey = rubyTypeNameSegments.size match {
-          case x if x == 1 =>
-            ""
-          case x if x > 1 =>
-            rubyTypeNameSegments.take(x - 1).mkString(".")
-        }
-
-        if gemsMap.contains(namespaceKey) then gemsMap.update(namespaceKey, gemsMap(namespaceKey) :+ rubyType)
-        else gemsMap.put(namespaceKey, List(rubyType))
-      }
+      val gemsMap = genGemToRubyTypesMap(rubyTypes)
 
       val typesFile = File(s"${dir.pathAsString}/$gem.mpk")
```

Comment:
```suggestion
      val typesFile = dir / s"$gem.mpk"
```

Reply:
Doesn't seem to work, I'm using `better-files` so not sure if that's why

![image](https://github.com/joernio/joern-type-stubs/assets/11736140/4907a083-c8cd-4cb0-b524-567e611dbe00)

Reply:
`/` is a function, make it as my suggestion without the `File`